iati/tests/test_resources.py

```diff
@@ -50,193 +177,492 @@ class TestResourceFolders(object):
+        (iati.resources.get_organisation_schema_paths, iati.resources.FILE_SCHEMA_ORGANISATION_NAME)
+    ])
+    def func_and_name(self, request):
+        """Return a named tuple containing a function to generate the paths for a type of Schema, plus the name of the Schema."""
+        output = collections.namedtuple('output', 'func schema_name')
+        return output(func=request.param[0], schema_name=request.param[1])
+
+    @pytest.fixture(params=[
         iati.resources.get_all_schema_paths,
         iati.resources.get_activity_schema_paths,
         iati.resources.get_organisation_schema_paths
     ])
-    def test_find_schema_paths_file_extension(self, std_ver_minor_mixedinst_valid_fullsupport, create_schema_path_function):
-        """Check that the correct file extension is present within file paths returned by get_all_*schema_paths functions."""
-        paths = create_schema_path_function(std_ver_minor_mixedinst_valid_fullsupport)
+    def schema_path_func_all(self, request):
+        """Return a function that returns a list of paths for Schema resources."""
+        return request.param
 
-        for path in paths:
-            assert path[-4:] == iati.resources.FILE_SCHEMA_EXTENSION
+    def test_get_schema_paths_minor_known(self, std_ver_minor_mixedinst_valid_known, func_and_name):
+        """Test getting a list of Org or Activity Schema paths. The requested version is known by pyIATI."""
+        result = func_and_name.func(std_ver_minor_mixedinst_valid_known)
 
-    def test_schema_activity_string(self, std_ver_minor_inst_valid_known):
-        """Check that the Activity schema file contains content.
+        assert len(result) == 1
+        assert result[0] == iati.resources.create_schema_path(func_and_name.schema_name, std_ver_minor_mixedinst_valid_known)
 
-        Todo:
-            Change fixture to mixedinst.
+    def test_get_schema_paths_minor_unknown(self, std_ver_all_mixedinst_valid_unknown, schema_path_func_all):
+        """Test getting a list of Org or Activity Schema paths. The requested version is not known by pyIATI."""
+        result = schema_path_func_all(std_ver_all_mixedinst_valid_unknown)
 
-        """
-        path = iati.resources.create_schema_path('iati-activities-schema', std_ver_minor_inst_valid_known)
+        assert result == []
 
-        content = iati.utilities.load_as_string(path)
+    def test_get_schema_paths_independent(self, schema_path_func_all):
+        """Test getting a list of version-independent Org or Activity Schemas."""
+        result = schema_path_func_all(iati.version.STANDARD_VERSION_ANY)
+
+        assert result == []
+
+    def test_get_schema_paths_major_known(self, std_ver_major_uninst_valid_known, func_and_name):
+        """Test getting a list of Org or Activity Schema paths. The requested version is a known integer version. The list should contain paths for each supported minor within the major."""
```

Comment:
Docstrings in this class are clunky and a bit unclear.

Should look to split info across multiple lines rather than smushing the info onto a single line.